[CI] MemoryTrackingTDigestArraysTests testIntBulkSet failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #4235 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/cdxgisx7qqppg)
- [elasticsearch-periodic #4230 / openjdk21_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/yn3jjtpz3zvli)
- [elasticsearch-periodic #4230 / openjdk23_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/obcx5ofa2qjeq)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.search.aggregations.metrics.MemoryTrackingTDigestArraysTests.testIntBulkSet" -Dtests.seed=D7098FAD3010FB55 -Dtests.locale=mgh-MZ -Dtests.timezone=America/Monterrey -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testIntBulkSet'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.search.aggregations.metrics.MemoryTrackingTDigestArraysTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: max must be >= min: 0, -1
```

**Issue Reasons:**
- [main] 3 failures in test testIntBulkSet (0.5% fail rate in 550 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/server/src/test/java/org/elasticsearch/search/aggregations/metrics/MemoryTrackingTDigestArraysTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/metrics/MemoryTrackingTDigestArraysTests.java
@@ -91,7 +91,7 @@ public class MemoryTrackingTDigestArraysTests extends ESTestCase {
             assertThat(target.size(), equalTo(targetArraySize));
             assertThat(source.size(), equalTo(sourceArraySize));
 
-            int value = randomIntBetween(Integer.MIN_VALUE, Integer.MAX_VALUE);
+            int value = randomInt();
             for (int i = 0; i < sourceArraySize; i++) {
                 source.set(i, value);
             }

--- a/server/src/test/java/org/elasticsearch/search/aggregations/metrics/MemoryTrackingTDigestArraysTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/metrics/MemoryTrackingTDigestArraysTests.java
@@ -84,29 +84,29 @@ public class MemoryTrackingTDigestArraysTests extends ESTestCase {
     }
 
     public void testIntBulkSet() {
-        int initialSize = randomIntBetween(10, 1000);
-        int sourceArraySize = randomIntBetween(0, initialSize);
+        int targetArraySize = randomIntBetween(10, 1000);
+        int sourceArraySize = randomIntBetween(0, 1000);
 
-        try (TDigestIntArray array = intArray(initialSize); TDigestIntArray source = intArray(sourceArraySize)) {
-            assertThat(array.size(), equalTo(initialSize));
+        try (TDigestIntArray target = intArray(targetArraySize); TDigestIntArray source = intArray(sourceArraySize)) {
+            assertThat(target.size(), equalTo(targetArraySize));
             assertThat(source.size(), equalTo(sourceArraySize));
 
-            int value = randomInt();
+            int value = randomIntBetween(Integer.MIN_VALUE, Integer.MAX_VALUE);
             for (int i = 0; i < sourceArraySize; i++) {
                 source.set(i, value);
             }
 
-            int initialOffset = randomIntBetween(0, initialSize - sourceArraySize);
-            int sourceOffset = randomIntBetween(0, sourceArraySize - 1);
-            int elementsToCopy = randomIntBetween(1, sourceArraySize - sourceOffset);
+            int targetOffset = randomIntBetween(0, targetArraySize);
+            int sourceOffset = randomIntBetween(0, sourceArraySize);
+            int elementsToCopy = randomIntBetween(0, Math.min(sourceArraySize - sourceOffset, targetArraySize - targetOffset));
 
-            array.set(initialOffset, source, sourceOffset, elementsToCopy);
+            target.set(targetOffset, source, sourceOffset, elementsToCopy);
 
-            for (int i = 0; i < initialSize; i++) {
-                if (i < initialOffset || i >= initialOffset + elementsToCopy) {
-                    assertThat(array.get(i), equalTo(0));
+            for (int i = 0; i < targetArraySize; i++) {
+                if (i < targetOffset || i >= targetOffset + elementsToCopy) {
+                    assertThat(target.get(i), equalTo(0));
                 } else {
-                    assertThat(array.get(i), equalTo(value));
+                    assertThat(target.get(i), equalTo(value));
                 }
             }
         }
@@ -289,11 +289,11 @@ public class MemoryTrackingTDigestArraysTests extends ESTestCase {
     }
 
     public void testDoubleBulkSet() {
-        int initialSize = randomIntBetween(10, 1000);
-        int sourceArraySize = randomIntBetween(0, initialSize);
+        int targetArraySize = randomIntBetween(10, 1000);
+        int sourceArraySize = randomIntBetween(0, 1000);
 
-        try (TDigestDoubleArray array = doubleArray(initialSize); TDigestDoubleArray source = doubleArray(sourceArraySize)) {
-            assertThat(array.size(), equalTo(initialSize));
+        try (TDigestDoubleArray target = doubleArray(targetArraySize); TDigestDoubleArray source = doubleArray(sourceArraySize)) {
+            assertThat(target.size(), equalTo(targetArraySize));
             assertThat(source.size(), equalTo(sourceArraySize));
 
             double value = randomDoubleBetween(-Double.MAX_VALUE, Double.MAX_VALUE, true);
@@ -301,17 +301,17 @@ public class MemoryTrackingTDigestArraysTests extends ESTestCase {
                 source.set(i, value);
             }
 
-            int initialOffset = randomIntBetween(0, initialSize - sourceArraySize);
-            int sourceOffset = randomIntBetween(0, sourceArraySize - 1);
-            int elementsToCopy = randomIntBetween(1, sourceArraySize - sourceOffset);
+            int targetOffset = randomIntBetween(0, targetArraySize);
+            int sourceOffset = randomIntBetween(0, sourceArraySize);
+            int elementsToCopy = randomIntBetween(0, Math.min(sourceArraySize - sourceOffset, targetArraySize - targetOffset));
 
-            array.set(initialOffset, source, sourceOffset, elementsToCopy);
+            target.set(targetOffset, source, sourceOffset, elementsToCopy);
 
-            for (int i = 0; i < initialSize; i++) {
-                if (i < initialOffset || i >= initialOffset + elementsToCopy) {
-                    assertThat(array.get(i), equalTo(0.0));
+            for (int i = 0; i < targetArraySize; i++) {
+                if (i < targetOffset || i >= targetOffset + elementsToCopy) {
+                    assertThat(target.get(i), equalTo(0.0));
                 } else {
-                    assertThat(array.get(i), equalTo(value));
+                    assertThat(target.get(i), equalTo(value));
                 }
             }
         }

